main.js

```diff
@@ -8,7 +9,12 @@ import { initSync, __wasm } from "./module.js";
  */
 const init = async (input) => {
   if (typeof input === "undefined") {
-    input = readFileSync("./main.wasm");
+    const wasmPath = typeof __dirname === "undefined"
+      // in esm build path can be relative
+      ? "./main.wasm"
```

Comment:
If this is ESM, then we can be more explicit here, and use `new URL("./main.wasm", import.meta.url);`

Reply:
Brilliant!